### PR TITLE
Rework AMD CI pipeline to use pool AMD-GPU and disable more tests in order to enable it.

### PR DESCRIPTION
**Pull request by suffiank**
**Description**: Describe your changes.
Rework AMD CI pipeline to use pool AMD-GPU and disable more tests in order to enable it.

**Motivation and Context**
- Why is this change required? What problem does it solve?
A first step to enable AMD CI pipeline as PR gate. Some unit tests are disabled because Rocm coverage is not complete. I would say the plan is to investigate and re-enable these unit tests in stages. As of now, the target language models (e.g. BERT-L, GPT) have been validated working.

### CODE REVIEW

**Review by weixingzhang — DISMISSED** [review]

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
 
 name: 'orttraining_ci_$(Date:yyyyMMdd)_$(Rev:r)'
-
-pool: 'Linux-CPU-CorpNet'
-
-variables:
-- group: pai-variables
-- name: pai-user
-  value: onnxrt
+pool: 'AMD-GPU'
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:4]: can i get rid of Linux-CPU-CorpNet now? :)

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml [hunk 2]
```diff
@@ -1,33 +1,38 @@
...
+    --update \
+    --build_dir ./build \
+    --build \
+    --parallel 8 \
+    --build_wheel \
+    --skip_tests
+  displayName: 'Build onnxruntime'
+
+- script: |-
+    cd ./build/RelWithDebInfo &&\
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:35]: fyi, there is a workingDirectory parameter
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#script

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml (hunk 1 above)
Comment (suffiank) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:4]: O yes :)

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml (hunk 2 above)
Comment (suffiank) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:35]: OK, good to know. I think I'll incorporate on next revision (more to come as tests are brought back..)

**Review by weixingzhang — APPROVED** [review]

**Pull request merged**